### DISCUSSION TITLE
Invalid value provided. Expected Float or Null, provided String.

### DISCUSSION THREAD

**Discussion by DwayneSSM** [prisma/prisma#22341] [Q&A]
### Question

Hello,
I keep getting the following error for the same value

ERROR:
```
Invalid `.create()` invocation in
/Users/juliocortez/Workspace/nuxt-ds-test/server/api/shopos/index.post.ts:45:1

  42 } = shopQuery;
  43 
  44 await tx.vehicle
→ 45   .create({
         data: {
          ...
           query: {
             create: {
               ...
               feul_amount: "3000.0",
                                   ~~~~~~~~
               ...
             }
           }
         }
       })

Argument `fuel_amount`: Invalid value provided. Expected Float or Null, provided String.

```

the value is defined as below in the schema:
`fuel_amount                                Float?`

I have checked the type going in and have even used parsefloat() to double be sure it is indeed a number it works if the value is null but if it is a number or string containing a number such as "3000" or "2000" i get the error above.

3000.0 string <-- before conversion
3000 number <-- after

anyone i have an idea what is up?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS:  macOS
- Database: SQL Server
- Node.js version: v20.6.0

- Run `prisma -v` to see your Prisma version and paste it
- ```
prisma                  : 5.4.2
@prisma/client          : 5.7.0
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine ac9d7041ed77bcc8a8dbd2ab6616b39013829574 (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli ac9d7041ed77bcc8a8dbd2ab6616b39013829574 (at node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.4.1-2.ac9d7041ed77bcc8a8dbd2ab6616b39013829574
Default Engines Hash    : ac9d7041ed77bcc8a8dbd2ab6616b39013829574
Studio                  : 0.494.0
```

**Comment by ludralph** [comment]
Hi @DwayneSSM 👋 

Thank you for raising this question.

The error message indicates that Prisma is expecting a Float or Null value for the `fuel_amount` field, but it is receiving a String. Even though you are using parseFloat() to convert the string to a number, the error persists. Can you please share the schema and query and surrounding codes leading to the conversion? If possible you could share a minimal repository with instructions on how to reproduce so we could try reproducing this.

**Reply by wsanada** [reply]
When we pass a number as a string to a Decimal field, for example, it works perfectly.
If we send a statement directly to Postgres, SQLServer, whatever, it works fine too. I can't understand this unnecessary restriction from Prisma.

**Reply by DwayneSSM** [reply]
Only just now seeing this somehow. 
I got around it still using parseFloat but do not remember what change i made for it to work. I agree with wsanada, though. the restriction seem unnecessary.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
